src/Forms/FormMessage.php

```diff
@@ -92,6 +91,30 @@ public function setMessage(
         return $this;
     }
 
+    /**
+     * Appends a message to the existing message if the types match, overrides if not
+     */
+    public function appendMessage(
+        string $message,
+        string $messageType = ValidationResult::TYPE_ERROR,
+        string $messageCast = ValidationResult::CAST_TEXT,
+    ): static {
+        if (!empty($this->message)) {
+            $canBeMerged = ($messageType === $this->getMessageType() && $messageCast === $this->getMessageCast());
+            // Checks that the exact message string is not already contained before appending
+            if ($canBeMerged && strpos($this->message, $message) === false) {
```

Comment:
I feel like if it can't be merged, we should probably throw an exception rather than overriding the existing message.

Reply:
Yeah there's merit to that. Maybe a "force" flag that overrides if set, throws if not?

Reply:
Yeah that would be acceptable I think. At least then it's a clear and explicit decision developers are making